test/HW05Tests.hs

```diff
@@ -0,0 +1,56 @@
+module HW05Tests (allTests) where
+
+import qualified Data.ByteString.Lazy as BS
+import System.Directory
+import Test.Hspec
+
+import Week05.HW05
+
+allTests :: Spec
+allTests = describe "\nWeek05" $ do
+    describe "\nHaskell Bank" $ do
+        it "\n getSecret read a key" $ do
+            key <- getSecret "src/Week05/dog-original.jpg" "src/Week05/dog.jpg"
+            key `shouldSatisfy` (not . BS.null)
+        after_ deleteDecrypted $
+            it "\n create decrypted file" $ do
+                key <- getSecret "src/Week05/dog-original.jpg" "src/Week05/dog.jpg"
+                decryptWithKey key "src/Week05/victims.json"
+        after_ deleteDecrypted $
+            it "\n get bad transactions" $ do
+                key <- getSecret "src/Week05/dog-original.jpg" "src/Week05/dog.jpg"
+                decryptWithKey key "src/Week05/victims.json"
+                badTs <- getBadTs "src/Week05/victims.json" "src/Week05/transactions.json"
+                badTs `shouldSatisfy` (not . null)
+        after_ deleteDecrypted $
+            it "\n track the flow of money from bad transactions" $ do
+                key <- getSecret "src/Week05/dog-original.jpg" "src/Week05/dog.jpg"
+                decryptWithKey key "src/Week05/victims.json"
+                badTs <- getBadTs "src/Week05/victims.json" "src/Week05/transactions.json"
+                case badTs of
+                    Nothing -> error "No Transactions"
```

Comment:
Try avoiding error whenever possible. In hspec tests in particular, it's better to use [expectationFailure](https://hackage.haskell.org/package/hspec-expectations-0.8.4/docs/Test-Hspec-Expectations.html#v:expectationFailure)